gordo_components/builder/build_model.py

```diff
@@ -314,13 +320,39 @@ def set_seed(self, seed: int):
         np.random.seed(seed)
         random.seed(seed)
 
+    @staticmethod
+    def build_split_dict(X: pd.DataFrame, tssplit: TimeSeriesSplit) -> dict:
+        """
+        Get dictionary of cross-validation training dataset split metadata
+
+        Parameters
+        ----------
+        X: pd.DataFrame
+            The training dataset that will be split during cross-validation.
+        tssplit: sklearn.model_selection.TimeSeriesSplit
```

Comment:
I _think_ one could generalize this to any splitter

Reply:
It does for those that return indices. For example, this wouldn't work with [train_test_split](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html)

Reply:
No, any cross validation splitter.. `TimeSeriesSplit`, `KFold..` blah blah

Reply:
https://scikit-learn.org/stable/modules/classes.html#splitter-classes

Reply:
Right, so there are "Splitter Classes" and "splitter functions". So should I just create a custom type hint class for sklearn splitter classes?

Reply:
`sklearn.model_selection._split.BaseCrossValidator`